qa/L0_backend_python/response_sender/response_sender_test.py

```diff
@@ -517,12 +517,16 @@ def test_decoupled_one_response_pre_and_on_return(self):
         #       using `py_future.result()` with error hangs on exit.
 
     # Decoupled model return 1 response and send 1 response.
+    @unittest.skip("Response factory segmentation fault, see the TODO comment.")
     def test_decoupled_one_response_on_and_post_return(self):
         # Note: The returned response will send an error response and complete final
         #       flag. The response sender is not yet closed due to the model is still
         #       executing. Then, sending a response and final flag will make their way
         #       to the frontend, but the request is completed at the frontend, so they
         #       are not sent to the client.
+        # TODO: The error response will close the response factory, but not the response
```

Comment:
* If the user choses to send an error response will it close the response factory too or this only happens for unexpected errors (e.g. raising an exception).

     * If it only happens for exceptions, would it be possible to close the response sender for all the requests in the batch to avoid the segfault?

Reply:
> If the user choses to send an error response will it close the response factory too or this only happens for unexpected errors (e.g. raising an exception).

I think this happens only for unexpected errors (e.g. raising an exception), because the error of an error response is encapsulated within the response itself. The stub does not know/check if the response contains an error.

> If it only happens for exceptions, would it be possible to close the response sender for all the requests in the batch to avoid the segfault?

Good idea! In this case, the all response factories will be closed anyway, so it make sense to mark all response senders closed (if not already closed) to prevent any more use on the factory pointer.

Reply:
Updated: [Enable test_decoupled_one_response_on_and_post_return](https://github.com/triton-inference-server/server/pull/7292/commits/f7a2e56ef1b455d0fe1696950b97a033e87a0bb7)